.github/workflows/client-verify.yaml

```diff
@@ -39,7 +39,7 @@ jobs:
 
       - name: Install tox
         run: |
-          pip install tox
+          pip install tox==4.16.0
```

Comment:
Would it be worth installing the requirements-dev.txt file instead of a separate pip install here (and in the other actions)? Otherwise we'll have a lot of places that we have to update each time we want to upgrade a library...

Reply:
The tox is installed by the actions for different components.  Where should the requirements.txt file put?  Repository root?  If so, should it be named like tox_version.txt instead of requirements.txt?  @psschwei WDYT?

Reply:
my initial thought was use the dev requirements file for client / docs / gateway / proxy ... that would cover everything except the kubernetes test...

maybe one requirements.txt in the main `test/` directory? 

not really a great answer here unfortunately... I'm fine with what ever you think is best @akihikokuroda  (including pinning the versions in the action)